### PR TITLE
Directly handle rate limiting using `client.rate_limit.resets_in`.

### PR DESCRIPTION
**Pull request by ioquatix**
We discussed some issues here, and at least when I tried, it, I ran into rate limiting issues.

https://github.com/github-changelog-generator/github-changelog-generator/discussions/965#discussioncomment-816764

This PR tries to respect GitHub's rate limiting.

### CODE REVIEW

**Comment by olleolleolle** [comment]
Ugh I was so curious about the test results, and it was styling issues that stopped the build. I should flip those around, check style after running tests.

File: lib/github_changelog_generator/octo_fetcher.rb [hunk 1]
```diff
@@ -456,10 +455,21 @@ def iterate_pages(client, method, *arguments, parent: nil, **options)
...
     # @return [Object] returns exactly the same, what you put in the block, but wrap it with begin-rescue block
     # @param [Proc] block
-    def check_github_response(&block)
-      Retriable.retriable(retry_options, &block)
+    def check_github_response
+      yield
     rescue MovedPermanentlyError => e
       fail_with_message(e, "The repository has moved, update your configuration")
+    rescue Octokit::TooManyRequests => e
+      resets_in = client.rate_limit.resets_in
```
Comment (olleolleolle) [inline lib/github_changelog_generator/octo_fetcher.rb:463]: 🔍 I learned! resets_in (a time) is even provided by Octokit, thanks for finding this out!

**Review by olleolleolle — APPROVED** [review]
👍 This lets us use Octokit in a more precise way.

File: lib/github_changelog_generator/octo_fetcher.rb (hunk 1 above)
Comment (ioquatix) [reply lib/github_changelog_generator/octo_fetcher.rb:463]: You are most welcome!

**Comment by ioquatix** [comment]
Thanks for handling this!

**Comment by aelam** [comment]
it seems that It does not work well as before.

**Comment by ioquatix** [comment]
@aelam what is the problem?

**Comment by aelam** [comment]
@ioquatix Sorry for late, after I waited a long time, it worked

**Comment by Mizommz** [comment]
> We discussed some issues here, and at least when I tried, it, I ran into rate limiting issues.
> 
> https://github.com/github-changelog-generator/github-changelog-generator/discussions/965#discussioncomment-816764
> 
> This PR tries to respect GitHub's rate limiting.

File: lib/github_changelog_generator/octo_fetcher.rb (hunk 1 above)
Comment (Mizommz) [reply lib/github_changelog_generator/octo_fetcher.rb:463]: https://androidonair.withgoogle.com/events/android-security?talk=session-2

**Pull request merged**